### PR TITLE
feat(git): add support for auth headers for clones

### PR DESCRIPTION
**Pull request by dmccaffery**
* add authorization header detection when cloning repositories
* add support for setting global configuration
* update all manually applied configs to use config methods on ```IGitRepositoryInitialized``` ```IGitRepositoryCanInitialize``` interfaces
* add support for product name in ```GetDocsMetadata```
* update build agent name to VSTS for TFS (they both use the same VSTS build agents now)

### CODE REVIEW

File: src/AM.Condo.Abstractions/IO/IGitRepositoryFactory.cs [hunk 1]
```diff
@@ -24,13 +24,16 @@ public interface IGitRepositoryFactory
         /// <param name="uri">
         /// The URI of the git repository that should be cloned.
         /// </param>
+        /// <param name="authorization">
+        /// The authorization header to use for access the repository.
```
Comment (tiffanywang3) [inline src/AM.Condo.Abstractions/IO/IGitRepositoryFactory.cs]: *accessing

Comment (dmccaffery) [reply src/AM.Condo.Abstractions/IO/IGitRepositoryFactory.cs]: *to access

Comment (sjk07) [reply src/AM.Condo.Abstractions/IO/IGitRepositoryFactory.cs]: rookie mistake @dmccaffery

Comment (tiffanywang3) [reply src/AM.Condo.Abstractions/IO/IGitRepositoryFactory.cs]: I changed it to accessing because you use present participles elsewhere.

Comment (dmccaffery) [reply src/AM.Condo.Abstractions/IO/IGitRepositoryFactory.cs]: @sjk07 we can't all be winners :feelsgood: 

@tiffanywang3 fixed

**Review by tiffanywang3 — APPROVED** [review]

**Pull request merged**